frontend/apps/reader/modules/readerdogear.lua

```diff
@@ -22,17 +22,22 @@ function ReaderDogear:init()
 end
 
 function ReaderDogear:resetLayout()
-    self[1].dimen.w = Screen:getWidth()
+    local new_screen_width = Screen:getWidth()
+    if new_screen_width == self._last_screen_width then return end
+    local new_screen_height = Screen:getHeight()
+    self._last_screen_width = new_screen_width
+
+    self[1].dimen.w = new_screen_width
     if Device:isTouchDevice() then
         self.ges_events = {
             Tap = {
                 GestureRange:new{
                     ges = "tap",
                     range = Geom:new{
-                        x = Screen:getWidth()*DTAP_ZONE_BOOKMARK.x,
-                        y = Screen:getHeight()*DTAP_ZONE_BOOKMARK.y,
-                        w = Screen:getWidth()*DTAP_ZONE_BOOKMARK.w,
-                        h = Screen:getHeight()*DTAP_ZONE_BOOKMARK.h
+                        x = new_screen_width*DTAP_ZONE_BOOKMARK.x,
```

Comment:
That code path will only be executed when there is a change in screen size. Since it's setting everything up with the new size, I thought it makes more sense to prepend the new prefix.

Before this change, it was doing this initialization multiple times when creating a readerui for a new file, which is kinda horrible...